src/lib/image/image.c

```diff
@@ -52,16 +52,25 @@
 
 struct image_object singularity_image_init(char *path, int open_flags) {
     struct image_object image;
+    char *urip;
     char *real_path;
 
     if ( path == NULL ) {
         singularity_message(ERROR, "No container image path defined\n");
         ABORT(255);
     }
 
+    urip = strstr(path, "://");
+
+    if (urip != NULL) {
+        *(urip + 3) = '\0';
+        singularity_message(ERROR, "Image paths beginning with %s are not supported\n", path);
```

Comment:
Do you mean to be using `urip` in the format string instead of `path` - from the setting of the `\0` above it looks like maybe this is intended?

Reply:
I think using path is right ... though I don't think the output will be what's expected. This looks like it's just being sure that `path` isn't a URI (correct me if I'm wrong here).

But the message should print out the entire thing... like if somehow `docker://ubuntu:latest' got passed here, then the error message would be:

`Image paths beginning with docker://ubuntu:latest are not supported` ... if `urip` was used, the output should be:
`Image paths beginning with :// are not supported`

Since it's doing a check for `://` I'd change the output to something like:

`A URI location is invalid as an image path`

Reply:
Sorry, brain fade - I think this does make sense here, as the error should be specific:

`Image paths beginning with docker:// are not supported` which this will do.